### PR TITLE
Support Dell v4 API

### PR DESCRIPTION
**Pull request by djjudas21**
* Update the warranty fact to use Dell's v4 API (new fact taken from [jhaals/warranty](https://github.com/jhaals/jhaals-warranty))
* Update param to default to latest yumrepo

The change to the fact also requires a new API key, so this should be tagged a breaking change / major release.

### CODE REVIEW

**Comment by djjudas21** [comment]
Hold fire on this, I'm still working some issues out

File: manifests/params.pp [hunk 1]
```diff
@@ -34,7 +34,7 @@
           $omsa_url_args_indep = 'osname=el$releasever&basearch=$basearch&native=1&dellsysidpluginver=$dellsysidpluginver'
           $omsa_url_args_specific = 'osname=el$releasever&basearch=$basearch&native=1&sys_ven_id=$sys_ven_id&sys_dev_id=$sys_dev_id&dellsysidpluginver=$dellsysidpluginver'
           $omsa_version = $::productname ? {
-            default    => 'Linux_Repository_15.04.00',
+            default    => 'Linux_Repository_17.03.00',
```
Comment (raphink) [inline manifests/params.pp]: Does this belong with this PR?

Comment (djjudas21) [inline manifests/params.pp]: Probably should be separate. It isn't required to support the v4 API

**Comment by raphink** [comment]
Thanks for the PR. What issues are still left?

**Comment by djjudas21** [comment]
It is now fully tested and working on my CentOS 7 installation, and I've updated the README to explain how to manage the API key.

**Comment by raphink** [comment]
Great, thank you!

File: manifests/params.pp (hunk 1 above)
Comment (raphink) [inline manifests/params.pp]: Could you leave it then please?

File: manifests/init.pp [hunk 2]
```diff
@@ -18,4 +19,10 @@
   validate_absolute_path($customplugins)
 
   validate_string($check_warranty_revision)
+
+  validate_string($api_key)
+
+  class { 'dell::warranty':
```
Comment (raphink) [inline manifests/init.pp]: I prefer to use absolute class names. Could you use `::dell::warranty` here please?

Comment (djjudas21) [inline manifests/init.pp]: Well spotted. Fixed!

File: manifests/params.pp (hunk 1 above)
Comment (djjudas21) [inline manifests/params.pp]: You want me to move this commit into a different PR?

Comment (raphink) [inline manifests/params.pp]: Come to think of it, I think it'll be fine, never mine. I'll just squash and merge.

**Comment by djjudas21** [comment]
Great, thanks. Can you tag a new release, please, and upload to Puppet Forge?

**Comment by raphink** [comment]
Done already, it's building right now ;-)

**Comment by raphink** [comment]
And it's released
